src/magic.cpp

```diff
@@ -623,11 +623,11 @@ bool spell::is_valid_target( valid_target t ) const
     return type->valid_targets[t];
 }
 
-bool spell::is_valid_target( const tripoint &p ) const
+bool spell::is_valid_target( const Creature &caster, const tripoint &p ) const
 {
     bool valid = false;
     if( Creature *const cr = g->critter_at<Creature>( p ) ) {
-        Creature::Attitude cr_att = cr->attitude_to( g->u );
+        Creature::Attitude cr_att = cr->attitude_to( caster );
```

Comment:
As you have changed it here: `g->u` is still used a few lines down in this very same function. You may want to add a comment if it's intentional that `g->u` is used and not `caster`.